From d216fcd05862fd7e4bb9395e2c96a0f24e63a548 Mon Sep 17 00:00:00 2001
From: Tom Stellard <tstellar@redhat.com>
Date: Wed, 17 Feb 2021 12:50:08 -0800
Subject: [PATCH] Include refstring.h from system includedir

---
 libcxxabi/src/stdlib_stdexcept.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcxxabi/src/stdlib_stdexcept.cpp b/libcxxabi/src/stdlib_stdexcept.cpp
index 4a464e48893d..c3d14b224186 100644
--- a/libcxxabi/src/stdlib_stdexcept.cpp
+++ b/libcxxabi/src/stdlib_stdexcept.cpp
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "../../libcxx/src/include/refstring.h"
+#include <libcxx-internal/refstring.h>
 #include "stdexcept"
 #include "new"
 #include <cstdlib>
-- 
2.27.0

